code/procs/jobprocs.dm

```diff
@@ -396,6 +409,34 @@
 				if(SL.len > 0)
 					src.set_loc(pick(SL))
 
+			if (src.traitHolder && src.traitHolder.hasTrait("pilot"))		//Has the Pilot trait - they're drifting off-station in a pod. Note that environmental checks are not needed here.
+				var/turf/pilotSpawnLocation = null
+				
+				#ifdef UNDERWATER_MAP										//This part of the code executes only if the map is a water one.
+				while(!pilotSpawnVerif(pilotSpawnLocation))					//Trying to find a valid spawn location.
+					pilotSpawnLocation = locate(rand(1, world.maxx), rand(1, world.maxy), 5)
+				if (pilotSpawnLocation)										//Sanity check.
+					src.set_loc(pilotSpawnLocation)
+				var/obj/machinery/vehicle/tank/minisub/V = new/obj/machinery/vehicle/tank/minisub/pilot(pilotSpawnLocation)
+				#else														//This part of the code executes only if the map is a space one.
+				var/pilotZ = rand(2, (world.maxz - 1))
+				var/validZ = FALSE
+				while (!validZ)
+					if (isrestrictedz(pilotZ))
+						pilotZ = ++pilotZ % (world.maxz - 1)
+						continue
+					validZ = TRUE
```

Comment:
I appreciate the sentiment of trying to think about the future when there might be more unrestricted z-levels but this could also be all replaced with just
```suggestion
				var/pilotZ = pick(3, 5)
```
now.

Reply:
Was allowed to keep the old code.